src/FakeItEasy/Creation/CastleDynamicProxy/CastleDynamicProxyGenerator.cs

```diff
@@ -19,13 +21,15 @@ internal class CastleDynamicProxyGenerator
         private static readonly ProxyGenerator ProxyGenerator = new ProxyGenerator();
         private readonly CastleDynamicProxyInterceptionValidator interceptionValidator;
 
+#if FEATURE_SECURITY_PERMISSIONS
         [SuppressMessage("Microsoft.Performance", "CA1810:InitializeReferenceTypeStaticFieldsInline", Justification = "No field initialization.")]
         static CastleDynamicProxyGenerator()
         {
 #pragma warning disable 618
             AttributesToAvoidReplicating.Add(typeof(SecurityPermissionAttribute));
 #pragma warning restore 618
         }
+#endif

```

Comment:
I have never understood anything about the `AttributesToAvoidReplicating`, but presumably if we don't have FEATURE_SECURITY_PERMISSIONS, neither does the Castle.Core we're consuming, so the point is moot?